modules/lidar_oscillation.py

```diff
@@ -0,0 +1,51 @@
+"""
+Representing a LiDAR Oscillation
+"""
+
+from . import lidar_detection
+
+
+class LidarOscillation:
+    """
+    Class to represent a collection of LiDAR readings that make up an oscillation.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(
+        cls, readings: list[lidar_detection.LidarDetection]
+    ) -> "tuple[bool, LidarOscillation | None]":
+        """
+        Create a new LidarOscillation object from a list of LidarReading objects.
+        """
+
+        return True, LidarOscillation(cls.__create_key, readings)
+
+    def __init__(
+        self, class_private_create_key: object, readings: list[lidar_detection.LidarDetection]
+    ) -> None:
+        """
+        Private constructor, use create() method to instantiate.
+        """
+        assert class_private_create_key is LidarOscillation.__create_key, "Use the create() method"
+
+        self.readings = readings
+        angles = [reading.angle for reading in readings]
+
+        if angles:
```

Comment:
since we checked list is empty, we can assume there are valid angle readings so we can remove this check